Source/Core/OrientedBoundingBox.js

```diff
@@ -0,0 +1,302 @@
+        Ellipsoid,
+        Intersect,
+        Plane,
+        Rectangle,
+        CesiumMath,
+        Matrix3) {
+    "use strict";
+
+    /**
+     * Creates an instance of an OrientedBoundingBox.
+     * An OrientedBoundingBox model of an object or set of objects, is a closed volume (a cuboid), which completely contains the object or the set of objects.
+     * It is oriented, so it can provide an optimum fit, it can bound more tightly.
+     * @alias OrientedBoundingBox
+     * @constructor
+     *
+     * @param {Cartesian3} [center=Cartesian3.ZERO] The center of the box.
+     * @param {Matrix3} [halfAxes=Matrix3.ZERO] The three orthogonal half-axes of the bounding box.
+     *                                          Equivalently, the transformation matrix, to rotate and scale a 2x2x2
+     *                                          cube centered at the origin.
+     *
+     * @see OrientedBoundingBox.fromBoundingRectangle
+     * @see BoundingSphere
+     * @see BoundingRectangle
+     * @see ObjectOrientedBoundingBox
+     *
+     * @example
+     * // Create an OrientedBoundingBox using a transformation matrix, a position where the box will be translated, and a scale.
+     * var center = new Cesium.Cartesian3(1,0,0);
+     * var halfAxes = Cesium.Matrix3.clone(Cesium.Matrix3.fromScale(new Cartesian3(1.0, 3.0, 2.0)));
+     *
+     * var obb = new Cesium.OrientedBoundingBox(center, halfAxes);
+     */
+    var OrientedBoundingBox = function(center, halfAxes) {
+        /**
+         * The center of the box.
+         * @type {Cartesian3}
+         * @default {@link Cartesian3.ZERO}
+         */
+        this.center = Cartesian3.clone(defaultValue(center, Cartesian3.ZERO));
+        /**
+         * The transformation matrix, to rotate the box to the right position.
+         * @type {Matrix3}
+         * @default {@link Matrix3.IDENTITY}
+         */
+        this.halfAxes = Matrix3.clone(defaultValue(halfAxes, Matrix3.ZERO));
+    };
+
+    var scratchCartesian1 = new Cartesian3();
+    var scratchCartesian2 = new Cartesian3();
+    var scratchCartesian3 = new Cartesian3();
+
+    var scratchRotation = new Matrix3();
+    /**
+     * Computes an OrientedBoundingBox from a BoundingRectangle.
+     * The BoundingRectangle is placed on the XY plane.
+     *
+     * @param {BoundingRectangle} boundingRectangle A bounding rectangle.
+     * @param {Number} [rotation=0.0] The rotation of the bounding box in radians.
+     * @param {OrientedBoundingBox} [result] The object onto which to store the result.
+     * @returns {OrientedBoundingBox} The modified result parameter or a new OrientedBoundingBox instance if one was not provided.
+     *
+     * @example
+     * // Compute an object oriented bounding box enclosing two points.
+     * var box = Cesium.OrientedBoundingBox.fromBoundingRectangle(boundingRectangle, 0.0);
+     */
+    OrientedBoundingBox.fromBoundingRectangle = function(boundingRectangle, rotation, result) {
```

Comment:
Hmm, maybe mark this as `@private` until we know we need it.  Or just remove it since we have the other file.